airbyte-server/src/main/java/io/airbyte/server/handlers/SourceHandler.java

```diff
@@ -213,11 +213,15 @@ public void deleteSource(final SourceRead source)
     final var workspaceIdRequestBody = new WorkspaceIdRequestBody()
         .workspaceId(source.getWorkspaceId());
 
-    connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody)
+    final List<UUID> uuidsToDelete = connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody)
         .getConnections().stream()
         .filter(con -> con.getSourceId().equals(source.getSourceId()))
         .map(ConnectionRead::getConnectionId)
-        .forEach(connectionsHandler::deleteConnection);
+        .toList();
+
+    for (final UUID uuidToDelete : uuidsToDelete) {
+      connectionsHandler.deleteConnection(uuidToDelete);
+    }
```

Comment:
You could instead call `.forEach` instead of calling `toList` and then iterating over that list.

Reply:
It is done on purpose in order to have the exception thrown by `deleteConnection` be thrown by the method (deleteConnection starts throwing in this PR).